backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteNonRootBranch.java

```diff
@@ -97,6 +105,12 @@ void setUpstreamBranch(BaseGitMacheteBranch givenUpstreamBranch) {
     return Option.of(customAnnotation);
   }
 
+  @Override
+  @Pure
```

Comment:
Whoops.. this method is `@SideEffectFree`, but isn't `@Deterministic` (a different instance is returned every time by`Option.of` for a non-null value); thus it isn't `@Pure`. Checker allowed for that coz purity annotations are by default **trusted**, not **checked** (unlike most others).

Reply:
But actually, looking at the uses... let's leave the `@Pure` annotation everywhere in `Option`-returning methods (3 places in this PR AFAICS); just leave a comment about the shortcut you're taking

Reply:
I removed it (and other occurrences) completely coz they are not used - i resolved problem by getting optional from getter to variable and then do `isDefined()` and if is, then `get()`